OCC.Service.Webhost/Repositories/PersonRepository.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using OCC.Data;
+using OCC.Service.Webhost.Services;
+using OCC.Service.Webhost.Tools;
+using Person = OCC.Service.Webhost.Services.Person;
+
+namespace OCC.Service.Webhost.Repositories
+{
+    public class PersonRepository
+    {
+        private readonly OCCDB _dbContext;
+
+        public PersonRepository(OCCDB dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public int RegisterPerson(Person person)
+        {
+            var p = new OCC.Data.Person();
+            Mapper.CopyProperties(person, p);
+
+            _dbContext.People.Add(p);
+            _dbContext.SaveChanges();
+
+            return p.ID;
+        }
+
+        public Person Login(Person person)
+        {
+            Person dcAttendee = default(Person);
+
+            var bcAttendee =
+                _dbContext.People
+                    .SingleOrDefault(p =>
+                        p.Email == person.Email &&
+                        p.PasswordHash == person.PasswordHash);
+
+            if (bcAttendee != null)
+            {
+                dcAttendee = new Person();
+                Mapper.CopyProperties(bcAttendee, dcAttendee);
+            }
+
+            return dcAttendee;
+        }
+
+        public Person FindPersonByEmail(string email)
+        {
+            Person dcAttendee = default(Person);
+
+            var bcAttendee = _dbContext.People.Where(p => p.Email == email)
+                .SingleOrDefault();
+
+            if (bcAttendee != null)
+            {
+                dcAttendee = new Person();
+                Mapper.CopyProperties(bcAttendee, dcAttendee);
+            }
+            return dcAttendee;
+        }
+
+        public void ResetPassword(string emailAddress, string temporaryPassword, string temporaryPasswordHash)
+        {
+            if (String.IsNullOrEmpty(emailAddress))
```

Comment:
This was basically just pulled out from the svc file into it's own repository with only the dbContext changed to remove the using and instead use the InRequestScope dbContext. This is the current existing functionality and I wouldn't want to start changing that around without putting tests in place first.